ddtrace/contrib/pytest/plugin.py

```diff
@@ -102,6 +104,11 @@ def pytest_runtest_protocol(item, nextitem):
         span.set_tag(test.SUITE, item.module.__name__)
         span.set_tag(test.TYPE, SpanTypes.TEST.value)
 
+        # Parameterized test cases will have a `callspec` attribute attached to the pytest Item object.
+        if getattr(item, "callspec", None):
+            params = {"parameters": item.callspec.params, "metadata": {}}
+            span.set_tag(test.PARAMETERS, json.dumps(params))
```

Comment:
This will work fine for all simple Python objects `int`, `str`, `dict` and `list` but will fail for other objects. What should we do in this case?

Reply:
Here's the solution I found:

`json.dumps()` has an optional keyword arg `default=TypeError`, any non-builtin object can be represented using their `__dict__` implementation, or if this isn't possible (in the case of functions), we can use the alternative `repr()`. 

WDYT?

Reply:
unless it is necessary/part of the specification to do so, I doubt doing any kind of intelligent object traversal is going to be worth it.

that is `json.dumps(params, default=repr)` is probably good enough?

Reply:
Part of the CI Apps specifications were to provide inner values of objects if possible, and if not possible then to default to a string representation of the object.